### ISSUE TITLE
Getting errors from extension not from the command line

### ISSUE THREAD

**Issue by daiyam** [Spence-S/vscode-linter-xo#5]
Hi, 

The code:
```
setTimeout(() => {
	console.log('test');
}, 1000);
```

The errors:
```
Unsafe call of an `any` typed value. (@typescript-eslint/no-unsafe-call) XO(@typescript-eslint/no-unsafe-call) [1, 1]
Unsafe member access .log on an `any` value. (@typescript-eslint/no-unsafe-member-access) XO(@typescript-eslint/no-unsafe-member-access) [2, 2]
Unsafe call of an `any` typed value. (@typescript-eslint/no-unsafe-call) XO(@typescript-eslint/no-unsafe-call) [2, 2]
```

I'm getting those errors from the extension but not from the command line...
I don't have any settings/rules in `xo.options`.

**Comment by Spence-S** [comment]
@daiyam interesting. Are you by chance using a multi-root workspace? If you have a repo I can look at I'd be happy to figure out what's going on. I actually think xo.options may be broke right now, so it will use the default xo rules, and resolve whatever you have configured in your folder.

**Comment by daiyam** [comment]
After isolating the example in a simple project, the issue might come from directly XO. Here the projects: [tests-xo.zip](https://github.com/Spence-S/vscode-linter-xo/files/6894428/tests-xo.zip).
There are 2 projects:
- `cmd-nok-ext-nok`: the command line and the extension are returning the same errors
- `cmd-ok-ext-nok`: no errors from the command line, the extension is giving the same errors

The differences are:
- `cmd-nok-ext-nok`: 1 `tsconfig.json` with `"rootDir": "src"`
- `cmd-ok-ext-nok`: 2 `tsconfig.json`:
  - at the root,  the only listed file is`package.json`
  - at `src`, `"rootDir": "."` with a reference to the first one (used to do `import from '../package.json'`)

Anyway, it's me or there is really an issue with the code?

**Comment by daiyam** [comment]
Hum, not sure what I did... the command line is working for both projects in [tests-xo.zip](https://github.com/Spence-S/vscode-linter-xo/files/6894428/tests-xo.zip). (the extension is giving errors)
So the issue is there.

**Comment by Spence-S** [comment]
Thank you for reporting this and giving the test repos. It's a bit bizarre to me what is happening here, but I _think_ it's actually an issue with xo. 

When I run from the command line - `npx xo --print-config=src/test.ts` 
I can clearly see I _should_ be getting the errors (the ones the extension shows). 

the following is in the printed configs:
```json
		"@typescript-eslint/no-unsafe-member-access": [
			"error"
		],
		"@typescript-eslint/no-unsafe-call": [
			"error"
		],
```

However, as you said, running xo from the command line gives no output. 

I also cannot figure out why the extension is getting it right, because when I run the same code from the extension in the test dir, so that I lint the files programatically I still get no results. Going to open an issue in XO. and will leave this open here in the mean time...

**Comment by daiyam** [comment]
Glad to help. The extension is useful but I found the bug on the first project with XO. I was going a bit crazy about those `no-unsafe` until I found out it was a bug...

**Comment by Spence-S** [comment]
yeah - xo should be showing those as errors, but if you don't want them reported (correctly) by the extension, just turn off the rules in your config. In package.json would be. 

```json
 "xo": {
    "rules": {
      "@typescript-eslint/no-unsafe-call": "off",
      "@typescript-eslint/no-unsafe-member-access": "off"
    }
  }
```

**Comment by daiyam** [comment]
Well, I think those errors shouldn't be reported for that code. The code is useless but not unsafe.

**Comment by Spence-S** [comment]
@daiyam Ok you are correct - I figured out the cause of this but it will need to be fixed on xo side. The problem is that eslint ts parser is trying to resolve types on the side of the extension, based on the generated [parserOptions.project](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/parser#parseroptionsproject) typescript eslint option.

Not sure when/how to fix this, but I will leave this open until it is or I have a workaround for the extension. In the meantime, it's probably best to turn off rules which depend on local types, such as the 2 mentioned above here and probably other `no-unsafe` rules.

Basically, I need a way to tell xo in the extension to resolve the types in your directory, not in the extension directory, and currently there isn't a way to do that. 

Will update with more information when I have it...

**Comment by daiyam** [comment]
Ok. Thx

**Comment by Spence-S** [comment]
xo 0.43.0 is released now - can you confirm if you are still having issues? Your lintstaged file you may need to pass a --cwd flag to the xo command.

**Comment by Spence-S** [comment]
@daiyam if you are still having issues - open a new ticket on original repo and it can be addressed there.
